packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Program.cs

```diff
@@ -0,0 +1,258 @@
+            }
+
+            var config = args[0];
+            var profile = args[1];
+            var userData = args[2];
+
+            Connection _connection = new ConnectionBuilder().WithLogging().Build();
+            var portingAssistantPortingConfiguration = System.Text.Json.JsonSerializer.Deserialize<PortingAssistantPortingConfiguration>(File.ReadAllText(config));
+            string metricsFolder = Path.Combine(userData, "telemetry-logs");
+            Directory.CreateDirectory(metricsFolder);
+            TelemetryConfiguration teleConfig = new TelemetryConfiguration{
+              InvokeUrl = portingAssistantPortingConfiguration.PortingAssistantMetrics["InvokeUrl"].ToString(),
+              Region = portingAssistantPortingConfiguration.PortingAssistantMetrics["Region"].ToString(),
+              ServiceName = portingAssistantPortingConfiguration.PortingAssistantMetrics["ServiceName"].ToString(),
+              Description = portingAssistantPortingConfiguration.PortingAssistantMetrics["Description"].ToString(),
+              LogsPath = metricsFolder,
+              LogFilePath = Path.Combine(metricsFolder, $"portingAssistant-telemetry-{DateTime.Today.ToString("yyyyMMdd")}.log"),
+              MetricsFilePath = Path.Combine(metricsFolder, $"portingAssistant-telemetry-{DateTime.Today.ToString("yyyyMMdd")}.metrics"),
+              Suffix = new List<string>(){".log", ".metrics"}
+            };
+            var lastReadTokenFile = Path.Combine(teleConfig.LogsPath, "lastToken.json");
+            string prefix = portingAssistantPortingConfiguration.PortingAssistantMetrics["Prefix"].ToString();
+            var client = new HttpClient();
+
+            // Create a timer and set an interval.
+            var logTimer = new System.Timers.Timer();
+            logTimer.Interval = Convert.ToDouble(portingAssistantPortingConfiguration.PortingAssistantMetrics["LogTimerInterval"].ToString());
+            // Hook up the Elapsed event for the timer. 
+            // logTimer.Elapsed += OnTimedEvent;
+            logTimer.Elapsed += (source, e) => OnTimedEvent(source, e, teleConfig, lastReadTokenFile, client, profile, prefix);
+
+            // Have the timer fire repeated events (true is the default)
+            logTimer.AutoReset = true;
+
+            // Start the timer
+            logTimer.Enabled = true;
+
+            // LogWatcher logWatcher = new LogWatcher(teleConfig, profile, prefix);
+            // logWatcher.Start();
+            _connection.Listen();
+        }
+
+      private class PortingAssistantPortingConfiguration
+        {
+            public PortingAssistantConfiguration PortingAssistantConfiguration { get; set; }
+            public Dictionary<string, object> PortingAssistantMetrics { get; set; }
+        }
+      
+      private static void OnTimedEvent(Object source, System.Timers.ElapsedEventArgs e, TelemetryConfiguration teleConfig, string lastReadTokenFile, HttpClient client, string profile, string prefix )
```

Comment:
Ok for now, but we should definitely try and move the upload logic into a separate class. Would be best if we could do it in this iteration. It would help in decoupling the Log Upload logic